samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bridge_storage_manager.h"
+
+namespace
+{
+template <class T> bool LoadSimpleData(PersistentStorageNode *node, T &data)
+{
+	T tempData;
+	size_t readSize = 0;
+
+	bool result = PersistentStorage::Instance().Load(node, &tempData, sizeof(T), readSize);
+
+	data = tempData;
+	return result;
+}
+
+PersistentStorageNode CreateIndexNode(uint8_t bridgedDeviceIndex, PersistentStorageNode *parent)
+{
+	char index[BridgeStorageManager::kMaxIndexLength + 1] = { 0 };
+
+	snprintf(index, sizeof(index), "%d", bridgedDeviceIndex);
+
+	return PersistentStorageNode(index, strlen(index), parent);
+}
+
+} /* namespace */
+
+bool BridgeStorageManager::StoreBridgedDevicesCount(uint8_t count)
+{
+	return PersistentStorage::Instance().Store(&mBridgedDevicesCount, &count, sizeof(count));
+}
+
+bool BridgeStorageManager::LoadBridgedDevicesCount(uint8_t &count)
+{
+	return LoadSimpleData(&mBridgedDevicesCount, count);
+}
+
+bool BridgeStorageManager::StoreBridgedDevicesIndexes(uint8_t *indexes, uint8_t count)
+{
+	if (!indexes) {
+		return false;
+	}
+
+	return PersistentStorage::Instance().Store(&mBridgedDevicesIndexes, indexes, count);
+}
+
+bool BridgeStorageManager::LoadBridgedDevicesIndexes(uint8_t *indexes, uint8_t maxCount, size_t &count)
```

Comment:
Hmm, it is because indexes is an address of array containing multiple indexes and count is a size of this array. I can understand it could be unclear based on API, so I added docs to all methods in `BridgeStorageManager` module.

Reply:
Fair enough.